src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -2420,129 +2523,159 @@ subroutine atm_advance_acoustic_step_work(nCells, nEdges, nCellsSolve, cellStart
            if (cell1 <= nCellsSolve .or. cell2 <= nCellsSolve ) then
 
 !DIR$ IVDEP
+              !$acc loop vector
               do k=1,nVertLevels
                  ru_p(k,iEdge) = dts*tend_ru(k,iEdge)
-              end do
-!DIR$ IVDEP
-              do k=1,nVertLevels
-                 ruAvg(k,iEdge) = ru_p(k,iEdge)                 
+                 ruAvg(k,iEdge) = ru_p(k,iEdge)
               end do
 
            end if ! end test for block-owned cells
 
         end do ! end loop over edges
+        !$acc end parallel
 
       end if ! test for first acoustic step
 
       if (small_step == 1) then  ! initialize here on first small timestep.
+         !$acc parallel default(present)
+         !$acc loop gang worker vector collapse(2)
          do iCell=cellStart,cellEnd
-            rtheta_pp_old(1:nVertLevels,iCell) = 0.0
+            do k=1,nVertLevels
+               rtheta_pp_old(k,iCell) = 0.0
+            end do
          end do
+         !$acc loop gang worker
+         do iCell=cellSolveStart,cellSolveEnd
+            !$acc loop vector
+            do k=1,nVertLevels
+               rho_pp(k,iCell) = 0.0
+               rtheta_pp(k,iCell) = 0.0
+               wwAvg(k,iCell) = 0.0
+               rw_p(k,iCell) = 0.0
+            end do
+            wwAvg(nVertLevels+1,iCell) = 0.0
+            rw_p(nVertLevels+1,iCell) = 0.0
+         end do
```

Comment:
I would suggest we leave this initialization code in the loop further down (around line 2576).

Reply:
I will do as asked, and keep the modifications to a minimum.

Though, it seems like bad form to have an if condition inside a loop that doesn't depend on variables in the loop - especially when the code block immediately above tests the same condition and has similar behavior.